roles/ironic-data/tasks/main.yml

```diff
@@ -1 +1,96 @@
 ---
+- name: install ironic controller services
+  upstart_service: name={{ item }}
+                   user=ironic
+                   cmd=/usr/local/bin/{{ item }}
+                   config_dirs=/etc/ironic
+  with_items:
+    - ironic-conductor
+
+- name: install ironic pxe driver prereqs
+  apt: pkg={{ item }} state=installed
+  with_items:
+    - ipmitool
+    - ipxe
+    - open-iscsi
+    - qemu-utils
+    - syslinux
+    - syslinux-common
+    - tftp
+    - tftpd-hpa
+
+- name: create pxe and ipxe directories
+  file: name={{item}}
+        state=directory
+        owner=ironic
+        group=ironic
+        mode=0755
+  with_items:
+      - "{{ ironic.tftpboot_path }}"
+      - "{{ ironic.tftpboot_path }}/pxelinux.cfg"
+      - "{{ ironic.httpboot_path }}"
+      - "{{ ironic.httpboot_path }}/pxelinux.cfg"
+      - "{{ ironic.tftpboot_path }}/master_images"
+
+- name: "override default pxelinux.0 source for Ubuntu >= 14.10"
+  set_fact:
+     syslinux_tftp_dir: '/usr/lib/PXELINUX'
+  when: ansible_distribution == 'Ubuntu' and ansible_distribution_version|version_compare('14.10', '>=')
+
+- name: check if pxelinux.0 exists
+  stat: path={{ ironic.tftpboot_path }}/pxelinux.0
+  register: pxe
+
+- name: copy pxelinux.0 to tftpboot
+  command: cp {{syslinux_tftp_dir}}/pxelinux.0 {{ ironic.tftpboot_path }}/
+  when: pxe.stat.exists == False and ansible_distribution == 'Ubuntu' and ansible_distribution_version|version_compare('14.10', '>=')
```

Comment:
This could be simplified with a `creates=` argument to the `command` module. The argument can point to the path, and then the task will be skipped if the content already exists. Saves you the previous task too.